src/models/IPerson.ts

```diff
@@ -0,0 +1,6 @@
+export interface IPerson {
```

Comment:
I would agree, the more we are using this type across our codebase, the more it makes sense to extract it to a separate location.

I totally agree with the coupling argument. The `AddToList` component should not be coupled to be used inside the `List` component. It should not care where it is used.

The question is rather, should the `AddToList` component be coupled to the general purpose `IPerson` interface? Or should it define its own type?

One could argue, that as a presentational component, it defines its own type. It renders four input fields no matter how the `IPerson` interface changes. Then if the IPerson interface changes, we wouldn't necessarily need to change the source code of the `AddToList` component, but could change how we pass the props to it in the `List`.

On the other hand, one could say that in this specific app, it makes sense that whenever we change the `IPerson` interface, the `AddToList` component should change with it.

I hope this made sense 😅